providers/kavenegar-novu-sms/src/lib/kavenegar-novu-sms.provider.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  ISmsOptions,
+  ISmsProvider,
+} from '@novu/stateless';
+import axios from 'axios';
+
+export class KavenegarNovuSmsSmsProvider implements ISmsProvider {
+  id = 'kavenegar-novu-sms';
+  channelType = ChannelTypeEnum.SMS as ChannelTypeEnum.SMS;
+  url: string;
+
+  constructor(
+    private config: {
+      apiKey?: string;
+      from?: string;
+    }
+  ) {
+    this.url =
+      'https://api.kavenegar.com/v1/' + config.apiKey + '/sms/send.json';
+  }
+
+  async sendMessage(
+    options: ISmsOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const params = {
+      sender: this.config.from,
+      message: options.content,
+      receptor: options.to,
+    };
+
+    const result = await axios.post(this.url, params, {
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+      },
+    });
```

Comment:
What's the reason behind using an API call instead the Kavenegar Node package ? Are them not aligned? Is the API more advanced than the Node package?

Reply:
Node API package Structure was not compatible with your response form.
Kavenegar node package, responsing results in function that limited scope for return on upper provider function. it is not  async or promis.

Kavenegar Node Sample code:
```js
api.Send({
        message: "Message text",
        sender: "10004346",
        receptor: "09123456789,09367891011"
    },
    function(response, status) {
        console.log(response);
        console.log(status);
    }) ;
```
This code will return ```VOID```
We need this format for response:
```js
res = await this.KavenegarSDK.send({
        message: "Message text",
        sender: "10004346",
        receptor: "09123456789,09367891011"
      });
```
maybe there was a solution, but my time is very limit for update that, so i selected easier way to programming provider.

Reply:
I understand the limited time but you can easily transform a callback function request into a promise, there are multiple methods of doing it. Link you a bit of well explained resources to do that: https://advancedweb.hu/how-to-convert-between-callbacks-and-promises-in-javascript/

If the alignment between Node package and API SDK is not a problem for the dependency management for us is easier to implement the Node package. 🙏🏻

Reply:
OK
Thanks
I am changing this logic
i,ll commit very soon

Reply:
Changed to promise